### ISSUE TITLE
Cannot complie 2.4.4 with nix 1.8

### ISSUE THREAD

**Issue by zanerock** [arangodb/arangodb#1263]
Not sure this is an issue with Arango, but I'm not sure what to do next to debug further.

I'm trying to compile the 2.4.4 release within Nix (on Linux/openSUSE 13.1, though that shouldn't matter). It finds the opensll library, but then complains about 'CRYPTO_get_ex_data'. I grepped the nix installed openssl libs (1.0.1k), and that function is defined (see below).

Is there another library I need to include in the build?

From build:
```
configure: --------------------------------------------------------------------------------
configure: CHECKING FOR OPENSSL
configure: --------------------------------------------------------------------------------
checking openssl/ssl.h usability... yes
checking openssl/ssl.h presence... yes
checking for openssl/ssl.h... yes
checking for CRYPTO_get_ex_data in -lcrypto... no
configure: error: please install the OpenSSL library
```

Check for CRYPTO function:
```
user@conveyor-wkst:~/playground/dogfoodsoftware.com/distro> find /nix/store/8nxzd6jkgcihyx26m246ycpnpivicpak-openssl-1.0.1k/ -type f | xargs grep CRYPTO_get_ex_data
/nix/store/8nxzd6jkgcihyx26m246ycpnpivicpak-openssl-1.0.1k/include/openssl/crypto.h:const CRYPTO_EX_DATA_IMPL *CRYPTO_get_ex_data_implementation(void);
/nix/store/8nxzd6jkgcihyx26m246ycpnpivicpak-openssl-1.0.1k/include/openssl/crypto.h:void *CRYPTO_get_ex_data(const CRYPTO_EX_DATA *ad,int idx);
/nix/store/8nxzd6jkgcihyx26m246ycpnpivicpak-openssl-1.0.1k/include/openssl/symhacks.h:#undef CRYPTO_get_ex_data_implementation
/nix/store/8nxzd6jkgcihyx26m246ycpnpivicpak-openssl-1.0.1k/include/openssl/symhacks.h:#define CRYPTO_get_ex_data_implementation	CRYPTO_get_ex_data_impl
Binary file /nix/store/8nxzd6jkgcihyx26m246ycpnpivicpak-openssl-1.0.1k/lib/libssl.so.1.0.0 matches
Binary file /nix/store/8nxzd6jkgcihyx26m246ycpnpivicpak-openssl-1.0.1k/lib/libcrypto.so.1.0.0 matches
```

**Comment by dothebart** [comment]
I've experienced similar things with the most recent .debs, they disable SSLv3 which 2.4 doesn't yet treat as optional; CRYPTO_get_ex_data also seems to be deprecated. Right now our configure searches for it; its not used in the code itself.
The quick trick is to edit m4/external.openssl and replace it by CRYPTO_set_id_callback ; then run make setup to regenerate configure and configure again.

**Comment by dothebart** [comment]
fixed with #989eb5fc6e17b129c31ce377ac7c34be5f337692
